src/player.ts

```diff
@@ -2380,6 +2381,25 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  /**
+   * Checks if the video has reached the configured endTime and pauses if necessary.
+   * Pauses only once and allows the user to continue playing afterward.
+   * @private
+   * @returns {void}
+   */
+  private _checkEndTime(): void {
+    const endTime = this._sources.endTime;
+    const onTimeUpdate = (): void => {
```

Comment:
one last thing, can you please move the `const onTimeUpdate` inside the if block in line 2398? it's better to create it only if we have `endTime` configured and all other conditions exist.